src/utils/static-eval.ts

```diff
@@ -1,29 +1,38 @@
+import { safeStringify } from './safe-stringify';
 import { EvaluatedValue, StaticValue, ConditionalValue, Node } from './types';
-import { URL } from 'url';
-type Walk = (node: Node) => EvaluatedValue;
+type Walk = (node: Node) => Promise<EvaluatedValue>;
 type State = { computeBranches: boolean; vars: Record<string, any> };
 
+const walkCache = new Map<string, EvaluatedValue>();
```

Comment:
We should probably use a LRU cache to avoid running out of memory. Right now, the number of entries is unbounded.